flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/enumerator/ContinuousSplitPlannerImpl.java

```diff
@@ -213,17 +213,12 @@ static Optional<Snapshot> startSnapshot(Table table, ScanContext scanContext) {
             "Start snapshot id not found in history: " + scanContext.startSnapshotId());
         return Optional.of(matchedSnapshotById);
       case INCREMENTAL_FROM_SNAPSHOT_TIMESTAMP:
-        long snapshotIdAsOfTime =
-            SnapshotUtil.snapshotIdAsOfTime(table, scanContext.startSnapshotTimestamp());
-        Snapshot matchedSnapshotByTimestamp = table.snapshot(snapshotIdAsOfTime);
-        if (matchedSnapshotByTimestamp.timestampMillis() == scanContext.startSnapshotTimestamp()) {
-          return Optional.of(matchedSnapshotByTimestamp);
-        } else {
-          // if the snapshotIdAsOfTime has the timestamp value smaller than the
-          // scanContext.startSnapshotTimestamp(),
-          // return the child snapshot whose timestamp value is larger
-          return Optional.of(SnapshotUtil.snapshotAfter(table, snapshotIdAsOfTime));
-        }
+        Snapshot matchedSnapshotByTimestamp =
+            SnapshotUtil.oldestAncestorAfter(table, scanContext.startSnapshotTimestamp());
+        Preconditions.checkArgument(
+            matchedSnapshotByTimestamp != null,
+            "Cannot find a snapshot after: " + scanContext.startSnapshotTimestamp());
+        return Optional.of(matchedSnapshotByTimestamp);
```

Comment:
@stevenzwu FYI.

Isn't this a behavior change? I don't think we should make a behavior change.

Reply:
The old one was to first find a snapshot equal to the `scanContext.startSnapshotTimestamp()`. If not, look for a snapshot older than `scanContext.startSnapshotTimestamp()`. The logic is consistent with `oldestAncestorAfter`.
It come from https://github.com/apache/iceberg/pull/6350#discussion_r1044901322

Reply:
This fix seems correct. 

Old logic actually has a little problem, because it checks `List<HistoryEntry>` (not the ancestor history of the current table snapshot).